src/escalus_stanza.erl

```diff
@@ -232,9 +236,19 @@ from(Stanza, Recipient) when is_binary(Recipient) ->
 from(Stanza, Recipient) ->
     setattr(Stanza, <<"from">>, escalus_utils:get_jid(Recipient)).
 
+
+-spec set_id(exml:element()) -> exml:element().
+set_id(Stanza) ->
+    set_id(Stanza, uuid_v4()).
+
 set_id(Stanza, ID) ->
     setattr(Stanza, <<"id">>, ID).
 
+-spec set_current_timestamp(exml:element()) -> exml:element().
+set_current_timestamp(Stanza) ->
+    Timestamp = integer_to_binary(os:system_time(microsecond)),
```

Comment:
`Timestamp` should be passed from the outside, since getting it from the runtime environment inside the function body makes the function untestable (it's not a pure function anymore, but relies on external world state).